pennylane/ops/qutrit/state_preparation.py

```diff
@@ -0,0 +1,112 @@
+"""
+# pylint:disable=abstract-method,arguments-differ,protected-access,no-member
+from pennylane import numpy as np
+from pennylane import math
+from pennylane.operation import AnyWires, StatePrep
+from pennylane.templates.state_preparations import QutritBasisStatePreparation
+from pennylane.wires import Wires, WireError
+
+state_prep_ops = {"QutritBasisState"}
+
+
+class QutritBasisState(StatePrep):
+    r"""QutritBasisState(n, wires)
+    Prepares a single computational basis state for a qutrit system.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 1
+    * Gradient recipe: None (integer parameters not supported)
+
+    .. note::
+
+        If the ``QutritBasisState`` operation is not supported natively on the
+        target device, PennyLane will attempt to decompose the operation
+        into :class:`~.TShift` operations.
+
+    Args:
+        n (array): prepares the basis state :math:`\ket{n}`, where ``n`` is an
+            array of integers from the set :math:`\{0, 1, 2\}`, i.e.,
+            if ``n = np.array([0, 1, 0])``, prepares the state :math:`|010\rangle`.
+        wires (Sequence[int] or int): the wire(s) the operation acts on
+
+    **Example**
+
+    >>> dev = qml.device('default.qutrit', wires=2)
+    >>> @qml.qnode(dev)
+    ... def example_circuit():
+    ...     qml.QutritBasisState(np.array([2, 2]), wires=range(2))
+    ...     return qml.state()
+    >>> print(example_circuit())
+    [0.+0.j 0.+0.j 0.+0.j 0.+0.j 0.+0.j 0.+0.j 0.+0.j 0.+0.j 1.+0.j]
+    """
+    num_wires = AnyWires
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = (1,)
+    """int: Number of dimensions per trainable parameter of the operator."""
+
+    @staticmethod
+    def compute_decomposition(n, wires):
+        r"""Representation of the operator as a product of other operators (static method). :
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+        .. seealso:: :meth:`~.BasisState.decomposition`.
+
+        Args:
+            n (array): prepares the basis state :math:`\ket{n}`, where ``n`` is an
+                array of integers from the set :math:`\{0, 1, 2\}`
+            wires (Iterable, Wires): the wire(s) the operation acts on
+
+        Returns:
+            list[Operator]: decomposition into lower level operations
+
+        **Example:**
+
+        >>> qml.QutritBasisState.compute_decomposition([1,0], wires=(0,1))
+        [QutritBasisStatePreparation(array([1, 0]), wires=[0, 1])]
+
+        """
+        return [QutritBasisStatePreparation(n, wires)]
+
+    def state_vector(self, wire_order=None):
+        """Returns a state-vector of shape ``(3,) * num_wires``."""
+        prep_vals = self.parameters[0]
+        if any(i not in [0, 1, 2] for i in prep_vals):
+            raise ValueError("QutritBasisState parameter must consist of 0, 1 or 2 integers.")
```

Comment:
Following up on my previous suggestion to do this check in the `__init__` method of this class, I would remove this check from here if you do make the suggested change as this check would then be redundant.